pydra/engine/specs.py

```diff
@@ -104,9 +114,11 @@ def hash(self):
             )
         inp_hash = hash_function(inp_dict)
         if hasattr(self, "_graph_checksums"):
-            return hash_function((inp_hash, self._graph_checksums))
-        else:
-            return inp_hash
+            inp_hash = hash_function((inp_hash, self._graph_checksums))
+        # setting inp_hash and changing the flag to False
+        super().__setattr__("inp_hash", inp_hash)
+        super().__setattr__("changed", False)
```

Comment:
not sure why these two lines use super.

Reply:
i wanted to avoid situation  when setting these attributes will set `changed` to `True`, but perhaps I should just use an `if ` statement in `self.__setattr__`